src/date/expand-pattern.js

```diff
@@ -38,6 +39,128 @@ return function( options, cldr ) {
 		);
 	}
 
+	function getBestMatchPattern( path, skeleton ) {
+		var availableFormats, ratedFormats, format, pattern;
+
+		pattern = cldr.main([ path, skeleton ]);
+
+		if ( skeleton && !pattern ) {
+			availableFormats = cldr.main([ path ]);
+			ratedFormats = [];
+
+			for ( format in availableFormats ) {
+				ratedFormats.push({
+					format: format,
+					pattern: availableFormats[format],
+					rate: compareFormats( skeleton, format )
+				});
+			}
+
+			ratedFormats = ratedFormats
+				.filter( function( format ) {
+					return format.rate > -1;
+				} )
+				.sort( function( formatA, formatB ) {
+					return formatA.rate - formatB.rate;
+				});
+
+			if ( ratedFormats.length ) {
+				pattern = augmentFormat( skeleton, ratedFormats[0].pattern );
+			}
+		}
+
+		return pattern;
+	}
+
+	function repeatStr( str, count ) {
+		var i, result = "";
+		for ( i = 0; i < count; i++ ) {
+			result = result + str;
+		}
+		return result;
+	}
```

Comment:
Let's move this function into `./src/util/string/repeat-str.js`. Maybe `[./src/util/string/pad.js](https://github.com/jquery/globalize/blob/master/src/util/string/pad.js)` could reuse it.